velox/dwio/common/tests/utils/FilterGenerator.h

```diff
@@ -242,12 +245,19 @@ class ColumnStats : public AbstractColumnStats {
 
   std::unique_ptr<Filter> makeRangeFilter(float startPct, float selectPct) {
     if (values_.empty()) {
-      return std::make_unique<velox::common::IsNull>();
+      if (allowNulls_) {
+        return std::make_unique<velox::common::IsNull>();
+      } else {
+        return std::make_unique<velox::common::BigintRange>(0, 0, false);
+      }
     }
     int32_t lowerIndex;
     int32_t upperIndex;
     T lower = valueAtPct(startPct, &lowerIndex);
     T upper = valueAtPct(startPct + selectPct, &upperIndex);
+    if (!allowNulls_) {
```

Comment:
Using allowNulls_ to detect if the caller wants to just get this range filter seems a bit hacky. Currently only the ParquetReaderBenchmark set it to false, but in the future other tests may set it to false. Then this function would never run the below lines. If we pass it from FilterSpec, the callers have more flexibility.